### ISSUE TITLE
Make generated key names configurable in ModelConverter

### ISSUE THREAD

**Issue by davenquinn** [marshmallow-code/marshmallow-sqlalchemy#276]
I am trying to control how fields are generated from my models to create more meaningful names for client applications (typically modifying the SQLAlchemy automatically-generated relationship names). `ModelConverter` currently doesn't have a way to control the generated field name for properties, and uses `prop.key` in all cases. I propose adding a `ModelConverter._key_for_property` method or similar.

I have implemented this in a subclass by inelegantly shimming `fields_for_model`:
```python
class SparrowConverter(ModelConverter):
    def _key_for_property(self, prop):
        # Default implementation.
        return prop.key
    def fields_for_model(self, model, **kwargs):
        # Shim fields_for_model so we can control the generated property names
        # Precompute new keys so we can use properties
        new_keys = {prop.key: self._key_for_property(prop)
                    for prop in model.__mapper__.iterate_properties}
        # Convert fields to models using library code.
        fields = super().fields_for_model(model, **kwargs)
        return {new_keys[k]: v for k, v in fields.items() if v is not None}
```
Other methods sometimes need to call `_key_for_property` where they currently use `prop.key`. A better implementation would actually change `fields_for_model` but I didn't want to mess with the other the logic in there.

This seems like it would increase conversion flexibility with minimum complexity penalty. I'm happy to submit a PR if this seems useful.

**Comment by sloria** [comment]
See https://marshmallow-sqlalchemy.readthedocs.io/en/latest/recipes.html#overriding-generated-fields . `auto_field` allows you to pass a column name, which may differ from the marshmallow field name. Does that meet your use case?

**Comment by davenquinn** [comment]
Hi – that functionality almost meets my use case, and I did look at those docs before structuring things as I did.

First, it's likely that I was being dense, but I couldn't figure out how to remove a non-desired field using this functionality (maybe you set to `None`?). Second, it's a poorer fit for controlling the model conversion process in general (for auto-generating Schemas from reflected tables). If you didn't have concrete classes for each Schema, you would still have to re-implement `fields_for_model` to create a field with your desired key while removing the improperly named field. At that point, you're better off just using ModelConverter's methods directly.

For reference, here is my implementation of `_key_for_property`:
```
    def _key_for_property(self, prop):
        # Only change columns for relationship properties
        if not isinstance(prop, RelationshipProperty):
            return prop.key

        # Relationship with a single local column should be named for its
        # column (less '_id' postfix)
        if hasattr(prop, 'local_columns') and len(prop.local_columns) == 1:
            col_name = list(prop.local_columns)[0].name
            if col_name != 'id':
                return trim_postfix(col_name, "_id")

        # One-to-many models should have the field
        # named after the local column
        if prop.secondary is None and not prop.uselist:
            return prop.key
        # Otherwise, we go with the name of the target model.
        return prop.target.name
```

**Comment by sloria** [comment]
Oh, I see--thanks for clarifying your use case. Feel free to send a PR exposing that hook.

**Comment by sloria** [comment]
Closed by https://github.com/marshmallow-code/marshmallow-sqlalchemy/pull/279 and https://github.com/marshmallow-code/marshmallow-sqlalchemy/pull/281

**Comment by sloria** [comment]
Released in 0.22.2
